app/models/user.rb

```diff
@@ -200,123 +229,72 @@ def can_make_challenges
      challenges. Please wait until they are completed before you create another."]
   end
 
-  def has_clearance?(permission_level)
+  def clearance?(permission_level)
     # for each group (devs, mods, admins) there is a bit in the permission_level
     # indicating whether that group has clearance.
     # a user has clearance if they are a member of any of the groups which has clearance.
     # if all bits are set, then everyone has clearance (including users not in any group);
     # if no bits are set, then no-one has clearance.
     everyone_clearance = permission_level == ~0
-    admin_clearance    = permission_level.anybits?(1 << 0)
-    mod_clearance      = permission_level.anybits?(1 << 1)
-    dev_clearance      = permission_level.anybits?(1 << 2)
+    admin_clearance = permission_level.anybits?(1 << 0)
+    mod_clearance = permission_level.anybits?(1 << 1)
+    dev_clearance = permission_level.anybits?(1 << 2)
 
     everyone_clearance ||
       (is_developer && dev_clearance) ||
       (is_moderator && mod_clearance) ||
-      (is_admin     && admin_clearance)
-  end
-
-  def submission_limit
-    max_dad_level = highest_level
-    return 1 if max_dad_level.zero?
-
-    return max_dad_level if max_dad_level < UNLIMITED_SUBMISSIONS_REQUIREMENT
-
-    -1
+      (is_admin && admin_clearance)
   end
 
   def profile_picture
-    profile_picture = avatar.thumb.url unless avatar.nil?
-
-    if profile_picture.blank?
-      return 'https://s3.us-east-2.amazonaws.com/do-art-daily-public/Default+User+Thumb.png'
-    end
-
-    profile_picture
+    avatar.thumb.url || 'https://s3.us-east-2.amazonaws.com/do-art-daily-public/Default+User+Thumb.png'
   end
 
+  ### Moderation ###
   def approve(reason, moderator)
-    ModeratorLog.create(user_id: moderator.id,
-                        target: self,
-                        action: "#{moderator.username} has approved #{username}.",
-                        reason: reason)
-    update_attribute(:approved, true)
+    create_mod_log(moderator, "#{moderator.username} has approved #{username}.", reason)
+    update_retain_password(approved: true)
```

Comment:
I don't see where update_retain_password is defined.

Reply:
Line  96 of users.rb
https://github.com/LastArtistStanding/last-artist-standing/blob/f3c9e5cde02e8dc10063c6d5b48537a153543ef6/app/models/user.rb#L96

Reply:
Also why is update_retain_password a thing?

Reply:
I **think** it's a workaround around the need to provide password and password_confirmation that isn't a dirty use of some attribute overwrite method Ruby provides. I think it's bad practice to do so.
Try asking Cougar, maybe.